packages/camera/camera_android_camerax/lib/src/android_camera_camerax.dart

```diff
@@ -452,13 +481,66 @@ class AndroidCameraCameraX extends CameraPlatform {
 
   /// Gets the supported step size for exposure offset for the selected camera in EV units.
   ///
-  /// Returns 0 when exposure compensation is not supported.
-  ///
   /// [cameraId] not used.
   @override
   Future<double> getExposureOffsetStepSize(int cameraId) async {
     final ExposureState exposureState = await cameraInfo!.getExposureState();
-    return exposureState.exposureCompensationStep;
+    final double exposureOffsetStepSize =
+        exposureState.exposureCompensationStep;
+    if (exposureOffsetStepSize == 0) {
+      // CameraX returns a step size of 0 if exposure compensation is not
+      // supported for the device.
+      throw CameraException(exposureCompensationNotSupported,
+          'Exposure compensation not supported for the device.');
+    }
```

Comment:
Looking for feedback on throwing an exception here/how to handle exposure compensation not being supported on the device.

Reply:
Yeah something like that! Returning a negative number or throwing an exception are really the only two options I can think of honestly, since we cannot return null.

Reply:
I suppose I lean towards a special negative return value, because it isn't an actual error (or at least CameraX doesn't think it is). But this is a weak preference, if you have any reason for preferring throwing an error instead.

Reply:
Do you know how the other platforms handle this? I'm not familiar with how `exposureOffsetStepSize` works, but it seems the app-facing package indicates that a zero should be returned when it is not supported: https://pub.dev/documentation/camera/latest/camera/CameraController/getExposureOffsetStepSize.html

I think we should just return 0 if the other platforms are doing this as well.

Reply:
Ah okay. I interpreted that as still suggesting that the camera supports exposure compensation, though, which it doesn't. I am okay returning 0 though it is technically different in that way, since one way or another the developer will find this out when attempting to set exposure index. @gmackall thoughts?

Reply:
I'm not sure if my terminology here is correct, but wouldn't returning a 0 here still indicate that exposure compensation IS supported on this device? Just that it is done without stepping? Or am I misunderstanding

Reply:
Ah, I see. If we want to return a `-1` to indicate that the feature is not supported, then we would need to add that to the app-facing documentation. Do you know how `camera_android` handles this? Does it throw an exception if exposure compensation is not supported or does it also return a default value?

Reply:
I don't think that documentation is indicating that the exposure step size can be negative, just that the exposure setting itself can be negative (and then explaining how that would interact with a given step size). I don't think it would make much sense for the step size itself to be negative, though I'm not 100% sure its impossible (the documentation isn't clear). @camsim99 is it possible to get confirmation that this value can't be negative, so we can safely use the negative value here to indicate it being unsupported?

Reply:
> I don't think that documentation is indicating that the exposure step size can be negative, just that the exposure setting itself can be negative (and then explaining how that would interact with a given step size).

This is correct. The formula is (Exposure compensation index) * (exposure offset step size) = (exposure offset), so that's just explaining that the offset can be negative if you set the compensation index to a negative value. The offset step size, at least for CameraX, will always be some positive value, unless un-supported.

Reply:
So, returning a negative value would be clear if we can modify the platform interface.

For `camera_android`, the [docs](https://developer.android.com/reference/android/hardware/camera2/CameraCharacteristics#CONTROL_AE_COMPENSATION_STEP) for the value it requests says that the value is available on all devices, so that's an interesting contrast. I assume CameraX just chooses not to return this value if it knows that trying to set an exposure offset will fail on the device. Would have to double check that though.

Reply:
Rereading the docs, I realized i misread the value that can be negative. Thanks for verifying that!

Reply:
Thanks for the research!

I changed it to return 0 since that is consistent with iOS and CameraX and added a note in the docs.

Reply:
After talking again with @gmackall, decided to change to -1 and modify the platform interface. I think it will be helpful to have a defined return value for unsupported exposure compensation going forward and help differentiate it from the other platforms, since iOS seems to not support step sizes by always returning 0 and it doesn't seem that exposure control is implemented on web or Windows.

@stuartmorgan let us know if you have any concerns about modifying the interface, though I believe this addition will help future-proof it. Windows could also, then, return -1.

Reply:
Going to land this for now since it's blocking me, but I can revisit this if needed and otherwise, will proceed with a PR for the platform interface.

Reply:
> @stuartmorgan let us know if you have any concerns about modifying the interface, though I believe this addition will help future-proof it.

What do you mean by "modify the platform interface"? If it's just a docs change that sounds reasonable given the discussion above, if it's a code change that's complicated.

Reply:
Just a docs change :) https://github.com/flutter/packages/pull/6182